### DISCUSSION TITLE
AVRCP Remote Control from Separate Controller ?

### DISCUSSION THREAD

**Discussion by jotkailay** [pschatzmann/ESP32-A2DP#226] [Q&A] [answered]
Thank you for this library. AVRCP is implemented in this library along with the required A2DP. In the bluetooth specification, specifically "2.3.1.1 Remote Control from Separate Controller" it shows how a remote control can operate separate from the device playing the music

In the ESP32 docs it mentioned `i2s_dac_mode_t` with the option `I2S_DAC_CHANNEL_DISABLE`, which turns off the internal DAC. Hopefully this would allow say a smartphone to connect to the ESP32-A2DP device, let the ESP32-A2DP device control the music, but keep the music playing from the smartphone. 

Is that possible using this ESP32-A2DP library? I cannot seem to make it work. The ESP32 device seems to steal the audio, with no way to turn off the internal DAC

thx

**Comment by pschatzmann** [comment, marked as answer]
I am not sure if this is even possible with the ESP32 functionality: Your mentioned i2s_dac_mode_t is for I2S and has nothing to do with A2DP or AVRCP and neither the A2DP, AVRCP documentation nor the Espressif examples mention this scenario.

What I can definitely confirm is, that this is __not supported by this library__!
Starting a A2DP Sink will always deactivate the sound output on your Mobile Phone (A2DP Source) and make it available at the Sink.

I suggest to start a discussion an an official ESP32 ADF forum.

**Comment by jotkailay** [comment]
Thank you *so much* for the quick response, and letting me know this library is not the answer. Saves me tons of time. 

`Starting a A2DP Sink will always deactivate the sound output on your Mobile Phone (A2DP Source) and make it available at the Sink.
`

That was my thinking as well but the Bluetooth spec 2.3.1.1 says Bluetooth AVRCP does support this mode, and AVRCP relies on A2DP to be enabled, however yes it seems unlikely it will work with the SINK turned on.

**Reply by jotkailay** [reply]
Just wanted to add a link to the Bluetooth spec that indicates this is possible:

[Bluetooth spec 2.3.1.1 Remote Control from Separate Controller](https://i.imgur.com/XCOJO0X.png)
